Sources/COWMacros/COWExpansionFactory.swift

```diff
@@ -218,30 +223,196 @@ internal class COWExpansionFactory<Context: MacroExpansionContext> {
   
   // MARK: Utilities
   
-  private func createDerivedStorageTypeDecl() -> StructDeclSyntax {
-    let members = appliedStructValidVarDecls.map {
-      MemberBlockItemSyntax(decl: $0)
+  // If `appliedStructDecl` conforms to Equatable and manually implements `==`,
+  // the derived storage must also conform to Equatable (otherwise we hit a
+  // compiler bug that breaks the build).
+  // https://forums.swift.org/t/macro-multiple-matching-functions-xyz-error-though-the-expanded-code-has-no-duplicate-functions-and-compiles/68700/4
+  // FIXME: remove the workaround when the compiler bug is fixed.
+  private func applyEquatableWorkaroundForDerivedStorage(
+    members: inout [MemberBlockItemSyntax],
+    protocols: inout [InheritedTypeSyntax]
+  ) {
+    // Bail out if derived storage doesn't conform to Equatable.
+    guard protocols.contains(where: {
+      guard let name = $0.type.identifier else {
+        return false
+      }
+      return equatableProtocolNames.contains(name) ||
+             hashableProtocolNames.contains(name)
+    }) else {
+      return
     }
     
+    // Bail out if `appliedStructDecl` does not manually implement `==`.
+    guard let nestingEqualFunc = appliedStructDecl.memberBlock.members
+      .lazy
```

Comment:
What's the purpose of this `.lazy`? The `nestingEqualFunc` is evaluated in place. Does `.lazy` help improve the performance?

Reply:
Without `.lazy`, a copy of the temporary result would be generated after each functional-style operation.

https://forums.swift.org/t/compiler-optimisations-for-functional-style-collection-algorithms/68291/19

Reply:
Amazing technique. But I still think that the purpose of this optimization needs to be commented in the source code. Or, newcomers to this repository may become confused about this.

Reply:
Wouldn't it be way too verbose? There are at least 3 `.lazy` usages already in this PR.

Reply:
The best I can do is to make an `extension` on `Sequence` and wrap `.lazy` + `.map()` into something like `.lazyMap()`. But this still feels strange. Actually I think this is a great opportunity for newcomers to know about this serious performance pitfall (the default behavior) and to adopt `.lazy` where appropriate. Thoughts?